paths.tex

```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
 
 \subsection{Consecutive Shortest Path Distance Algorithm}
 
-\begin{figure}[H]
-\begin{lstlisting}[frame=single]
-  Input : Graph (V,E,T), dist[][][] entries set to Inf
-  Output : dist[][][]
-
-  for (t,src) in {T(e) | e in E } x V:
-    st_e := (src,src)
-    T(e) := (t,t) # or (t, Inf)?
-    R := {st_e}
-    S := {}
-    dist[t][src][src] = 0
-
-    while R != {}:
-      pred_e := pop R
-      for e in edge_neighborhood(pred_e) - union S R:
-        t_int := intersect T(pred_e) T(e)
-        if (t_int != null):
-          dist[t][src][snd e] := min(dist[t][src][fst e] + 1,
-                                        dist[t][src][snd e])
-          insert(e,R)
-      S += pred_e
-    return dist
-\end{lstlisting}
-\caption{An algorithm to compute the lengths of all consecutive temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
```

Comment:
Again, both in this description and below, I am a fan of calling them complete/total mappings with default values of `\infty`. Since we're not dealing with the i/o bound, there's no need to distinguish it as a 3D array.